File: src/python/oop/VendingMachine/vending/states/return_change_state.py
from ..order_context import OrderContext
from .order_state import OrderState

class ReturnChangeState(OrderState):
    def __init__(self, context: OrderContext):
        super().__init__(context)

    def process(self):
        self.context._change = 0.0
        from .complete_order_state import CompleteOrderState
        self._next_state(CompleteOrderState(self.context), True)

    def state_info(self):
        if self.context.change <= 0:
            from .complete_order_state import CompleteOrderState
            self._next_state(CompleteOrderState(self.context), True)
            return

        self.context.vm._notify(f"Please peak up the change: '{self.context.change:.2f}'")

File: src/csharp/oop/VendingMachine/Vending/States/ReturnChangeState.cs
namespace VendingMachine.Vending.States;

public class ReturnChangeState(OrderContext context) : OrderStateBase(context)
{
    public override void Process()
    {
        Context.Change = Decimal.Zero;
        NextState(new CompleteOrderState(Context), true);
    }

    public override void StateInfo()
    {
        if (Context.Change <= Decimal.Zero)
        {
            NextState(new CompleteOrderState(Context), true);
            return;
        }

        Notify($"Please peak up the change: {Context.Change}");
    }
}
